python/pyspark/resource/profile.py

```diff
@@ -106,6 +106,13 @@ def __init__(
             self._executor_resource_requests = _exec_req or {}
             self._task_resource_requests = _task_req or {}
 
+            from pyspark.sql import is_remote
```

Comment:
I'm wondering if I need to move this part of building the remote resource file  into the function of `def id(self)`